padl/dumptools/inspector.py

```diff
@@ -323,6 +218,35 @@ def caller_frame() -> types.FrameType:
     return outer_caller_frameinfo(calling_module_name).frame
 
 
+def instructions_are_the_same(instr, target_instr, frame=None):
+    """Check if the disassebled instructions *instr* and *target_instr* are (basically) the same.
+
+    Mainly this checks if the *opname* and *argval* of the two instructions are the same,
+    additionally accounting for some subleties involving code object - argvals and LOAD-operators.
+    """
+    # for the LOAD_FAST operator, get the argval from the frame's locals
+    if (frame is not None and target_instr.opname == 'LOAD_FAST'
+            and target_instr.argval in frame.f_locals
+            and target_instr.argval in frame.f_code.co_varnames):
+        argval = frame.f_locals[target_instr.argval]
+    else:
+        argval = target_instr.argval
+    # if the instruction's argval contains code objects, compare the code
+    instr_argval = getattr(instr.argval, 'co_code', instr.argval)
+    target_argval = getattr(target_instr.argval, 'co_code', target_instr.argval)
+
+    if instr_argval not in (target_argval, argval):
+        return False
+
+    if instr.opname == target_instr.opname:
+        return True
+
+    # load operations can have many different name, accept if both are load ops regardless
+    # of which kind exactly
+    load_ops = ('LOAD_NAME', 'LOAD_FAST', 'LOAD_GLOBAL', 'LOAD_CONST', 'LOAD_DEREF')
+    return instr.opname in load_ops and target_instr.opname in load_ops
+
+
 def get_segment_from_frame(caller_frame: types.FrameType, segment_type, return_locs=False) -> str:
```

Comment:
I think there should be `assertion` for `segment_type`. Cause if there is typo in `segment_type` or something else is passed, things will fail pretty far down in the code.

Reply:
makes sense